File: src/main/java/com/spring/spring_init/oauth/userInfo/OAuth2UserInfoFactory.java
package com.spring.spring_init.oauth.userInfo;

import com.spring.spring_init.oauth.OAuthProvider;
import java.util.Map;

public class OAuth2UserInfoFactory {

    public static OAuth2UserInfo getOAuth2MemberInfo(
        OAuthProvider oAuthProvider,
        Map<String, Object> attributes
    ) {
        switch (oAuthProvider) {
            case GOOGLE -> {
                return new GoogleOAuth2UserInfo(attributes);
            }
            case KAKAO -> {
                return new KakaoOAuth2UserInfo(attributes);
            }
            case NAVER -> {
                return new NaverOAuth2UserInfo(attributes);
            }
//            default -> throw new UnsupportedProviderException();
            default -> throw new RuntimeException("UnsupportedProviderException");
        }
    }
}

File: src/main/java/com/spring/spring_init/oauth/userInfo/OAuth2UserInfoFactory.kt
package com.spring.spring_init.oauth.userInfo

import com.spring.spring_init.oauth.OAuthProvider

object OAuth2UserInfoFactory {

    @JvmStatic
    fun getOAuth2MemberInfo(
        oAuthProvider: OAuthProvider,
        attributes: Map<String, Any>
    ): OAuth2UserInfo {
        return when (oAuthProvider) {
            OAuthProvider.GOOGLE -> GoogleOAuth2UserInfo(attributes)
            OAuthProvider.KAKAO -> KakaoOAuth2UserInfo(attributes)
            OAuthProvider.NAVER -> NaverOAuth2UserInfo(attributes)
//            else -> throw UnsupportedProviderException()
        }
    }
}
